### PR TITLE
Breadcrumb: (feature) Optional current item a heading

### LINKED ISSUE #10286: Breadcrumb: Make the last item a Heading, for Accessibility?
#### Describe the feature that you would like added

- Option to make the last item a heading.

#### What component or utility would this be added to

- Breadcrumb

#### Have you discussed this feature with our team, and if so, who

- No, but I'll actively participate in this thread.

#### Additional context/screenshots

- I'm willing to make the code changes, but I'd like to consult y'all first.

### PR DESCRIPTION
**Pull request by maxwellred**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #10286 
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Gives the user a new prop, `currentItemHeading` (that must be paired with the isCurrentItem prop), that they can use to assign the last (a.k.a. current) item to a heading restricted to the following 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10445)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.base.tsx [hunk 1]
```diff
@@ -153,6 +153,15 @@ export class BreadcrumbBase extends BaseComponent<IBreadcrumbProps, any> {
...
+      <Tag className={this._classNames.item}>
+        <TooltipHost content={item.text} overflowMode={TooltipOverflowMode.Parent} {...this.props.tooltipHostProps}>
+          {item.text}
+        </TooltipHost>
+      </Tag>
+    );
+
     if (item.onClick || item.href) {
       return (
         <Link
```
Comment (jdhuntington) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.base.tsx]: Seems like this should take effect regardless of `href` existing or not on `item`. I would expect an item to always be rendered as an `h*` if I specified it.

File: packages/office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx [hunk 2]
```diff
@@ -99,6 +99,16 @@ export const BreadcrumbBasicExample: React.FunctionComponent = () => {
         ariaLabel="Breadcrumb with maxDisplayedItems set to 2 and overflowIndex set to 1"
         overflowAriaLabel="More links"
       />
+      <Label styles={labelStyles}>With current item rendered as a heading</Label>
+      <Breadcrumb
+        items={[
+          { text: 'Files', key: 'Files', onClick: _onBreadcrumbItemClicked },
+          { text: 'This is folder 1', key: 'd1', onClick: _onBreadcrumbItemClicked },
+          { text: 'This is folder 2', key: 'd2', isCurrentItem: true, currentItemHeading: 'h2' }
```
Comment (jdhuntington) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx]: Feels strange to have this special case only for when an item is current.

Can we modify `IBreadcrumbItem` to follow a more common pattern instead? `as` is a commonly supported prop that would make sense here:

```
<Breadcrumb
        items={[
          { text: 'Files', key: 'Files', onClick: _onBreadcrumbItemClicked },
          { text: 'This is folder 1', key: 'd1', onClick: _onBreadcrumbItemClicked },
          { text: 'This is folder 2', key: 'd2', isCurrentItem: true, as: 'h2' }
```

This has the added benefit of supporting headings for non-current items if someone wanted to do that:

```
<Breadcrumb
        items={[
          { text: 'Files', key: 'Files', onClick: _onBreadcrumbItemClicked, as: 'h1' },
          { text: 'This is folder 1', key: 'd1', onClick: _onBreadcrumbItemClicked, as: 'h2' },
          { text: 'This is folder 2', key: 'd2', isCurrentItem: true, as: 'h3' }
```

Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/examples/Breadcrumb.Basic.Example.tsx]: Definitely, I'll switch it over!

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.base.tsx (hunk 1 above)
Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.base.tsx]: @jdhuntington Agreed, I've adjusted the `<Link/>` component to also accept the heading prop so the user can have choose either headings with or without links but the heading will always be applied, thanks !
 
`<Link
          as={item.as ? item.as : 'a'}`

**Comment by HalcyonPride** [comment]
Thanks, @maxwellred! 😃

**Pull request merged**